### ISSUE TITLE
Multi-level inherit auth from parent doesn't share variable values

### ISSUE THREAD

**Issue by pierskarsenbarg** [a85/POSTMan-Chrome-Extension#9470]
**Describe the Issue**

> A high-level summary or description of the problem that you're facing.

I have a collection level variable called `Token` which contains an API key
The API I'm targeting uses the following as the authorisation header: `Authorization: token ${token}` (where `${token}` is the value in the variable.

I have a folder that inherits the auth from the collection.
In the folder I have an API request that inherits auth from the folder.

However, when I make a request, the variable isn't being passed down the levels.

**Steps to Reproduce**

> A list of steps that can be followed to recreate the issue.

1. Create a collection
2. Create a variable in the collection called `Token`
3. Set the Authorization to be an API Key with the following values:

```
Key: Authorization
Value: token {{Token}}
Add to: Header
```

4. Add a folder
5. Set the Authorization in the folder to `Inherit auth from parent`
6. Create a request within that folder
7. Set Authorization to be `Inherit auth from parent`
8. View headers for this request and show the hidden headers
9. The Auth header will just have the value `token` in it, without the variable value

**Screenshots or Videos**

> Please add screenshots or a short video to help visually explain the problem. 
> Ensure that the screenshots/videos do not contain any sensitive information such as API keys or Access Tokens. 

**Environment Information**

 - Platform Type: Native app
 - Postman Version: 8.0.2
 - Operating System: macOS 11.1

**Additional Context**

> Please add any other contextual information about the problem, that might be relevant and help explain the issue you're facing.

**Comment by vvs11** [comment]
@pierskarsenbarg I am not able to reproduce this with the steps you mentioned. Could you share a recording of what you're trying to do? Also, could you check if there is a value in "current value" section in the collection variable "Token"?

**Comment by pierskarsenbarg** [comment]
I've uploaded a screen recording here: https://youtu.be/gspmyk-qCkY 

(note that's not my actual API token)

**Comment by bluechiperic** [comment]
I am having a similar issue.  I have a an API collection where I am setting OAuth2 Authorization.  It worked before I upgraded to v8.03 (From v7).  Now, if I set the auth in the API, it works, but if I inherit from the collection it does not.

**Comment by aleksandrs-ledovskis** [comment]
Same issue as @bluechiperic has - after upgrade to Postman v8 the "Inherit auth from parent" (parent has "OAuth 2.0") is misbehaving badly. It seems to cache somehow(?) the old/expired `Bearer` token and stubbornly refuses to use new one from parent (no matter if root/collection's level or just a folder one level above some API).

When forcing "OAuth 2.0" for specific API entry, it works.

**Comment by bcc1** [comment]
Same issue as the 2 comments above - In the previous version you neede to do an 'update' after the 'use token'. This 'update' no longer exist (and is perhaps the reason that the childs auth. values are no longer updated)

**Comment by dan-lind** [comment]
Having the same issue, just wanted to encourage everyone to +1 the original post by @pierskarsenbarg for increased visibility.

**Comment by rooch84** [comment]
Same issue here. 

As a workaround, you can copy the token from the parent, then in the child, change the authorisation type from inherit to bearer token, then paste the token in the token field.

**Comment by leiha** [comment]
Same issue here
Fix it please it's really boring :(

**Comment by leiha** [comment]
in version 8.0.10 working properly again :) ty support

**Comment by aleksandrs-ledovskis** [comment]
I cannot confirm it working in v8.0.10 - the "Inherit auth from parent" still misbehaves.

[Release notes](https://www.postman.com/downloads/release-notes/) also don't speak of any relevant change in v8.0.10.

**Comment by leiha** [comment]
After update of 8.0.10
for me the bearer auth works properly again 
i don't try other thing 

> I cannot confirm it working in v8.0.10 - the "Inherit auth from parent" still misbehaves.
> 
> [Release notes](https://www.postman.com/downloads/release-notes/) also don't speak of any relevant change in v8.0.10.

**Comment by dan-lind** [comment]
> I cannot confirm it working in v8.0.10 - the "Inherit auth from parent" still misbehaves.
> 
> [Release notes](https://www.postman.com/downloads/release-notes/) also don't speak of any relevant change in v8.0.10.

Agree, "Inherit auth from parent" still doesn't work

**Comment by dan-lind** [comment]
Looks like works if you press "Save" after doing "Get new access token" -> "Use token" (found this workaround in above mentioned issue)

**Comment by kenozordi** [comment]
> Looks like works if you press "Save" after doing "Get new access token" -> "Use token" (found this workaround in above mentioned issue)

This works, after changing Auth to "inherit from parent" make sure to save all the updates made.

**Comment by MelanieGravel** [comment]
Still doesn't work for me in a multi-folder situation.

**Comment by amirsalaar** [comment]
i noticed I had not saved my changes on the parent collection

**Comment by chris-kck** [comment]
@DannyDainton @giridharvc7 I do believe that this is a bug as I have experienced this in my own collection. I have the Bearer token added to the **Collection** level, then the **Subfolder** inherits from the collection and the **request** under the subfolder also inherits from the parent - meaning the request inherits from the collection level.
|--Collection _(Bearer Authorization)_
|---Subfolder _(Inherit from parent)_
|-----Get Request _(Inherit from parent)_
Rarely, the behaviour works as expected to inherit the same auth type, but in most cases, the Authorization "hidden header" is not added to the request. When you change from inherit to Bearer token, the header is added back and request works well. I think this is the core of the issue ⚠️ .
Also, even if I try to inherit from one parent, the same issue occurs. Inheritance is a great idea but it needs to be consistent. My current workaround is to manually add the auth to each request 😢
Related: [#9763](https://github.com/postmanlabs/postman-app-support/issues/9763) [#9744](https://github.com/postmanlabs/postman-app-support/issues/9744)

Additional context: I am using a pre-request script to set the token environment variable. Seems the token is not being added dynamically, and when there is no token, the header dissapears.

**Comment by giridharvc7** [comment]
@chris-kck We've fixed this in the latest update, can you once restart the app/refresh your web app to see if things are working fine for you?

**Comment by chris-kck** [comment]
@giridharvc7 Thanks a lot for your timely response and dedication to postman. I used Help-> Clear Cache and Reload and it started working well.
What I noticed is that when the token variable is empty, the Authorization header is removed. When the collection pre-request script assigns the token value, the header is then added to the requests. Since I wanted to get a token before each request and replace the old token (different user login), I was initially deleting the token after the request (in the test section) but I am now deleting the token at the top of the pre-request. The header is retained this way and the token is updated as expected. (Hope this helps someone who encounters this issue)

Version 10.0.25
UI Version: 10.0.25-ui-221012-1025
Desktop Platform Version: 10.0.1 (10.0.1)
